### PR TITLE
Created upload_gatk_tool_docs, the updated version of clean_up_html_gatk

### PR DESCRIPTION
**Pull request by Marianie-Simeon**
### **Addresses** 
https://broadworkbench.atlassian.net/browse/DFE-432

### **Commit Summary**

Added Upload GATK Tools Docs, Unit Testing, Config File, Delete GATK Files

Updated requirements.txt and .gitignore

### CODE REVIEW

File: scripts/upload_gatk_docs/upload_gatk_tools_docs.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+            clean_file.write(format_html)
+
+    # Output files that was updated
+    print(str(len(copied_files)) + ' files copied and cleaned up')
+
+    # Return GATK Doc Dict
+    return article_dict
+
+
+def post_section(gatk_version, username, token, headers):
```
Comment (mmorgantaylor) [inline scripts/upload_gatk_docs/upload_gatk_tools_docs.py:122]: is this function actually posting a section or simply retrieving the url needed to post an article inside that section? or both? i think we said this last time but this function might need a more descriptive name

File: scripts/upload_gatk_docs/upload_gatk_tools_docs.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+                                 "description": "Tool documentation for GATK release " + gatk_version
+                                 }}
+
+        data = json.dumps(json_file)
+
+        try:
+            # create the folder
+            response = requests.post(url_Tool_Index, headers=headers, data=data, auth=(username + '/token', token))
+        except: 
+            exit("ERROR: Can't make section")
```
Comment (mmorgantaylor) [inline scripts/upload_gatk_docs/upload_gatk_tools_docs.py:149]: this error message isn't very descriptive - could you show the error message / actually raise the error instead?

File: scripts/upload_gatk_docs/upload_gatk_tools_docs.py [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    if updatefolder is None:
+        updatefolder = 'gatkdoc_update_' + gatkdoc_version
+    if not os.path.exists(updatefolder):
+        os.makedirs(updatefolder)
+
+    # Get each article
+    for article in article_dict.values():
+        print('in original article ' + article.file_name)
+        with open(article.local_path, 'r') as clean_file:
+            clean_file_data_soup = bs(clean_file, 'html.parser')
```
Comment (mmorgantaylor) [inline scripts/upload_gatk_docs/upload_gatk_tools_docs.py:182]: could you add a comment here explaining what this line is doing? additionally perhaps leave `BeautifulSoup` named that way when you import it, or name it something else more descriptive?

File: scripts/upload_gatk_docs/upload_gatk_tools_docs.py [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+            for url in clean_file_data_soup.findAll('a'):
+                original_url = url.get('href')
+                if original_url in article_dict.keys():
+                    fixed_url_text = original_url.replace(original_url, article_dict[original_url].url)
+                    fixed_url = clean_file_data_soup.new_tag("a", href=fixed_url_text)
+                    fixed_url.string = article_dict[original_url].title
+                    url.replace_with(fixed_url)
+
+            # Format updated GATK Doc HTML
+            format_html = clean_file_data_soup.prettify()
```
Comment (mmorgantaylor) [inline scripts/upload_gatk_docs/upload_gatk_tools_docs.py:197]: what kinds of formatting does `prettify` do? a comment here briefly explaining that would be helpful!

File: scripts/upload_gatk_docs/upload_gatk_tools_docs.py [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+    parser.add_argument('--gatkdoc_path', '-P', required=True, type=str, help='The path to Workspace/gatk/build/docs/gatkdoc/')
+    args = parser.parse_args()
+
+    # GATK version
+    gatk_version = args.gatk_version
+
+    # GATK docs file path
+    gatkdoc_path = args.gatkdoc_path
+
+    # TODO ReadME
```
Comment (mmorgantaylor) [inline scripts/upload_gatk_docs/upload_gatk_tools_docs.py:218]: TODONE :)

Comment (Marianie-Simeon) [reply scripts/upload_gatk_docs/upload_gatk_tools_docs.py:218]: lol, yes

**Pull request closed without merging**